workers/approov-token-binding/jwt-verifier.js

```diff
@@ -0,0 +1,127 @@
+const jwt = require("jsonwebtoken");
+const crypto = require('crypto')
+
+addEventListener('fetch', event => {
+  event.respondWith(handleRequest(event))
+})
+
+async function handleRequest(event) {
+  let token_payload
+  const token = event.request.headers.get('Approov-Token');
+
+  try {
+    token_payload = jwt.verify(token, Buffer.from(APPROOV_BASE64_SECRET, 'base64'), {algorithms: ['HS256'], ignoreExpiration: false});
+
+    const result = handleApproovTokenBindingCheck(event.request, token_payload)
+
+    event.waitUntil(log(result, event.request, token))
```

Comment:
Won't this waitUntil impact the performance massively? Do we need to log all the time?

Reply:
The opposite, it's to reduce the penalty incurred by logging. 

See Cloudflare [docs](https://developers.cloudflare.com/workers/about/tips/fetch-event-lifecycle/):

>waitUntil()
>
>Extends the lifetime of the event using a Promise passed into the function. Use this method to notify the runtime to wait for tasks, such as streaming and caching, that run longer than the usual time it takes to send a response. This is good for handling logging and analytics to third-party services, where you don't want to block the response.